types/block.go

```diff
@@ -30,7 +30,7 @@ func BlockGasLimit(ctx sdk.Context) uint64 {
 
 	// Otherwise get from the consensus parameters
 	cp := ctx.ConsensusParams()
-	if cp == nil || cp.Block == nil {
+	if cp.Abci == nil || cp.Block == nil {
```

Comment:
I guess `cp.Block == nil {` is correct here.

Reply:
Thanks! I removed all the nil check of ConsensusParams.Abci